pkg/reconciler/workload/namespace/namespace_reconcile.go

```diff
@@ -168,34 +184,60 @@ func (c *Controller) reconcileGVR(ctx context.Context, gvr schema.GroupVersionRe
 // to assign to. The condition of not being scheduled to a cluster will be reflected in
 // the namespace's status rather than by returning an error.
 func (c *Controller) ensureScheduled(ctx context.Context, ns *corev1.Namespace) error {
-	oldPClusterName := ns.Labels[ClusterLabel]
+	oldPCClusterNames := workloadClustersFromLabels(ns.Labels)
 
 	scheduler := namespaceScheduler{
 		getCluster:   c.clusterLister.Get,
 		listClusters: c.clusterLister.List,
 	}
-	newPClusterName, err := scheduler.AssignCluster(ns)
+	newAssignedClusters, err := scheduler.AssignClusters(ns)
 	if err != nil {
 		return err
 	}
 
-	if oldPClusterName == newPClusterName {
+	newPClusterNames := make(map[string]string)
+	for _, ac := range newAssignedClusters {
+		newPClusterNames[ac] = "Sync"
+	}
+
+	// Compare oldPclusterNames and NewPClusterName to see if we need to update the labels
+	// of the namespace.
+	if reflect.DeepEqual(oldPCClusterNames, newPClusterNames) {
 		return nil
 	}
 
 	klog.Infof("Patching to update cluster assignment for namespace %s|%s: %s -> %s",
-		ns.ClusterName, ns.Name, oldPClusterName, newPClusterName)
-	patchType, patchBytes := clusterLabelPatchBytes(newPClusterName)
+		ns.ClusterName, ns.Name, oldPCClusterNames, newPClusterNames)
+
+	patchType, patchBytes := namespaceSchedulingPatchBytes(oldPCClusterNames, newPClusterNames)
+
 	patchedNamespace, err := c.kubeClient.Cluster(logicalcluster.From(ns)).CoreV1().Namespaces().
 		Patch(ctx, ns.Name, patchType, patchBytes, metav1.PatchOptions{})
 	if err == nil {
 		// Update the label to enable the caller to detect a scheduling change.
-		ns.Labels[ClusterLabel] = patchedNamespace.Labels[ClusterLabel]
+		for key := range patchedNamespace.Labels {
+			if strings.HasPrefix(key, ClusterLabelPrefix) {
+				ns.Labels[key] = patchedNamespace.Labels[key]
```

Comment:
what does it mean to change `ns` here instead of returning `patchedNamespace` somehow?